pkg/pod/pod.go

```diff
@@ -98,6 +117,52 @@ func (c *Client) GetPods() (pods []*v1.Pod, err error) {
 	return listPods, nil
 }
 
+// ListPods returns list of all pods
+func (c *Client) ListPods() (pods []*v1.Pod, err error) {
+	var resList []*v1.Pod
+	listPods := c.PodWatcher.Informer().GetStore().List()
+
+	for _, pod := range listPods {
+		v1Pod, ok := pod.(*v1.Pod)
+		if !ok {
+			err := fmt.Errorf("could not cast %T to v1.Pod", pod)
+			klog.Error(err)
+			return nil, err
+		}
+
+		resList = append(resList, v1Pod)
+		klog.V(6).Infof("Appending pod ID: %s/%s to list.", v1Pod.UID, v1Pod.Name)
+	}
+
+	return resList, nil
+}
+
+// GetPod returns the pod object
+func GetPod(obj interface{}, i informersv1.PodInformer) (pod *v1.Pod, err error) {
+	currentPod, exists, err := i.Informer().GetStore().Get(obj)
+	if !exists || err != nil {
```

Comment:
Is there a potential timing condition here?  Will we ever get called while waiting for the pod to get created?